src/modules/common/components/NavigationTiles.js

```diff
@@ -38,131 +41,141 @@ const TilesRow = styled.View`
+    this._scrollview = ref
   };
 
-  onMomentumScrollEnd = ({ nativeEvent }) => {
-    this.setState({ xPosition: nativeEvent.contentOffset.x })
+  renderAnchorIcon = (name: string, language: string) => {
+    return <Icon
+    name={name}
+    style={{ transform: [{ scaleX: isRTL() ? -1 : 1 }] }}
+    onPress={this.onAnchorPress}
+    width={anchorWidth}
+    />
+  }
+
+  onMomentumScrollEnd = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const { nativeEvent } = event
+    const contentSizeDiff =
+      nativeEvent.contentSize.width - nativeEvent.layoutMeasurement.width
+
+    this.setState({
+      xPosition: nativeEvent.contentOffset.x,
+      contentSizeDiff: contentSizeDiff.toFixed(0)
+    })
   };
 
   render () {
     const {
       tiles,
       theme,
-      isWideScreen,
       navigationItemWidth,
       scrollViewWidth,
-      anchorWidth,
-      language
+      language,
+      isScrollable
     } = this.props
-    const isMoreThanThreeItems = tiles.length > SMALL_SCREEN_ITEMS_COUNT
 
     return (
       <TilesRow theme={theme}>
-        {(!isWideScreen && isMoreThanThreeItems) && (
-          <Icon
-            name='keyboard-arrow-left'
-            style={{ transform: [{ scaleX: isContentDirectionReversalRequired(language) ? -1 : 1 }] }}
-            onPress={this.onAnchorPress}
-            width={anchorWidth}
-          />
-        )}
+        {isScrollable && this.renderAnchorIcon('keyboard-arrow-left', language)}
         <ScrollView
           horizontal
-          ref={this.setRef}
+          ref={this.setScrollViewRef}
           contentContainerStyle={{
             flexGrow: 1,
-            justifyContent: 'space-between'
+            justifyContent: 'space-around'
           }}
           style={{ width: scrollViewWidth }}
           showsHorizontalScrollIndicator={false}
           pagingEnabled
+          scrollEnabled
           snapToInterval={navigationItemWidth}
           decelerationRate='fast'
           bounces={false}
           onMomentumScrollEnd={this.onMomentumScrollEnd}
           snapToAlignment='center'>
-          {tiles.map(tile => (
-            <NavigationTile
-              key={tile.path}
-              tile={tile}
-              theme={theme}
-              width={navigationItemWidth}
-            />
-          ))}
+          {tiles.map(tile => <NavigationTile key={tile.path} tile={tile} theme={theme} width={navigationItemWidth} />)}
         </ScrollView>
-        {!isWideScreen && isMoreThanThreeItems && (
-          <Icon
-            name='keyboard-arrow-right'
-            style={{ transform: [{ scaleX: isContentDirectionReversalRequired(language) ? -1 : 1 }] }}
-            onPress={this.onAnchorPress}
-            width={anchorWidth}
-          />
-        )}
+        {isScrollable && this.renderAnchorIcon('keyboard-arrow-right', language)}
       </TilesRow>
     )
   }
 }
 
 const NavigationTilesWithScrollableView = (props: {
```

Comment:
No particular reason for this. The idea is that I didn't see it would be ok to separate functionality. This will consume new component and no re-usable/no clear functionality component
Anyway; Update; I moved AnchorIcon to another file with onAnchorPress logic too.